### ISSUE TITLE
I love this library!

### ISSUE THREAD

**Issue by Jared-Sprague** [phretaddin/schemapack#20]
I just wanted to thank you for this awesome library!  We are using it on our multiplayer game and it's awesome!  Transfering the initial game state to the client with json was over 1.5MB when using json,  with schemapack it shrunk that to just a couple hundred KB! Such a performance improvement.  And It's soooo much easier that packing your own binary messages and couting bytelengths and offsets per type ugh!!!  They way you designed the schema definitions was just perfect for our needs.   You did it right!  I will always recommend using this for anyone using binary over websockets.  Thanks!

**Comment by crazyyi** [comment]
Did you put the schemas on the client side? I am also using this library but I find that I need to expose the entire communication protocol to the clients. Although the performance is improved it is more vulnerable for  potential network attacks.

**Comment by Jared-Sprague** [comment]
@crazyyi yes of course the schema is on the client, because this is a javascript game.  But this library doesn't make it any more or less secure,  because even if we didn't use schemapack the client would still be sending messages to the server via websocket using some protocol, which could be reverse engineered if someone was motivated enough no mater what. 

So we did take two steps to stop people protect against cheating.  
1.  We do all validation on the server. So if someone does figure out how to send wrong input to the server it wouldn't pass validation.
2.  We use uglify with mangle properties by prefix so all our schema files end up just being random characters.  so insread of _PLAYER._POSITION  becomes xf.df.

Number 1 is the most important for any multiplayer game.  Number 2 just makes it much harder to reverse engineer,  not guarrenteed defense, but a good deterent.

**Comment by crazyyi** [comment]
Thanks for your explanation. May I ask what uglify or obfuscate library do you use? I have tried uglify plugin for webpack but it doesn't work well with ES6 syntax.
